scene/gui/code_edit.cpp

```diff
@@ -2907,7 +2907,7 @@ void CodeEdit::_filter_code_completion_candidates_impl() {
 			offset = line_height;
 		}
 
-		if (in_string != -1) {
+		if (in_string != -1 && option.insert_text[0] != '^' && option.insert_text[0] != '&') {
```

Comment:
`CodeEdit` should be generic.

Reply:
It's because when you're in a NodePath or StringName, it still thinks you're in a string, and unquoting below doesn't work, so it creates unnecessary quotation marks.

`is_quoted()` also accomplishes this, should be generic enough.